test/simulator/timestamp/simulator_interface/simulator_interface.cpp

```diff
@@ -158,21 +104,82 @@ interface_rollback_transaction()
 }
 
 void
-interface_set_timestamp()
+interface_timestamp_transaction()
 {
 }
 
 void
-interface_timestamp_transaction()
+interface_session_query_timestamp()
 {
 }
 
 void
-interface_timestamp_transaction_uint()
+print_rules()
 {
 }
 
-void
-print_rules()
+int
+main(int argc, char *argv[])
 {
+    bool exit = false;
+
+    do {
+        std::cout << std::endl;
+
+        std::cout << "1: Session Management " << std::endl;
+        std::cout << "2: [Conn] set_timestamp()" << std::endl;
+        std::cout << "3: [Conn] query_timestamp()" << std::endl;
+        std::cout << "4: [Session] begin_transaction()" << std::endl;
+        std::cout << "5: [Session] commit_transaction()" << std::endl;
+        std::cout << "6: [Session] prepare_transaction()" << std::endl;
+        std::cout << "7: [Session] rollback_transaction()" << std::endl;
+        std::cout << "8: [Session] timestamp_transaction()" << std::endl;
+        std::cout << "9: [Session] query_timestamp()" << std::endl;
+        std::cout << "10: print rules for timestamps" << std::endl;
+        std::cout << "11: exit" << std::endl;
+
+        int choice = choose_num(1, 11, "timestamp_simulator >>");
+
+        try {
+            switch (choice) {
+            case 1:
+                interface_session_management();
+                break;
+            case 2:
+                interface_set_timestamp();
+                break;
+            case 3:
+                interface_conn_query_timestamp();
+                break;
+            case 4:
+                interface_begin_transaction();
+                break;
+            case 5:
+                interface_commit_transaction();
+                break;
+            case 6:
+                interface_prepare_transaction();
+                break;
+            case 7:
+                interface_rollback_transaction();
+                break;
+            case 8:
+                interface_timestamp_transaction();
+                break;
+            case 9:
+                interface_session_query_timestamp();
+                break;
+            case 10:
+                print_rules();
+                break;
+            case 11:
```

Comment:
You may want a `default` statement here.

Reply:
As discussed we check for this to ensure that the range is always between the values in the case statement. Hence, let's not worry about this.

Reply:
Updated in [199f738](https://github.com/wiredtiger/wiredtiger/pull/8417/commits/199f738890b1943ec1e0325002fbb7bdd5ecc0ba)